### PR TITLE
EVM integration

### PR DESCRIPTION
**Pull request by randy75828**

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -339,13 +339,14 @@ func NewEthermintApp(
 		maccPerms,
 		sdk.GetConfig().GetBech32AccountAddrPrefix(),
 	)
-	app.BankKeeper = bankkeeper.NewBaseKeeper(
+	bankKeeper := bankkeeper.NewBaseKeeper(
```
Comment (holyxiaoxin) [inline app/app.go:342]: this doesn't look like it's needed?

File: x/evm/types/params.go [hunk 2]
```diff
@@ -23,12 +23,11 @@ import (
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/ethereum/go-ethereum/core/vm"
-	"github.com/evmos/ethermint/types"
 )
 
 var (
 	// DefaultEVMDenom defines the default EVM denomination on Ethermint
-	DefaultEVMDenom = types.AttoPhoton
+	DefaultEVMDenom = "eswth"
```
Comment (holyxiaoxin) [inline x/evm/types/params.go:30]: any implications if this param is changed via gov?

File: app/app.go (hunk 1 above)
Comment (randy75828) [reply app/app.go:342]: this is because of our switcheo cosmos-sdk has hooks which requires a pointer

File: x/evm/types/params.go (hunk 2 above)
Comment (randy75828) [reply x/evm/types/params.go:30]: added guard in params.Validate() to prevent a change in evmdenom

**Pull request merged**